### PR TITLE
twiceredis optimization

### PR DESCRIPTION
**Pull request by tr3buchet**

### CODE REVIEW

**Comment by Cerberus98** [comment]
:+1: other than you sucking at tests

**Comment by roaet** [comment]
It appears other humans ran into this issue before I did.

**Comment by tr3buchet** [comment]
i hate everything

**Comment by asadoughi** [comment]
The travis-ci pr test reveals new failures after #422 was merged. Please check those out and rebase off master to update the travis-ci push test. Thanks.

**Comment by tr3buchet** [comment]
alright, got a few more to fix.

I'd be delighted for some help testing this live. I don't know what your process is for using netdev or whatever, last I heard there were some perf tests reaping the whole environment

**Comment by roaet** [comment]
No. Perf is waiting for shared IPs.

Sent from my iPhone

On Jul 29, 2015, at 1:13 PM, Trey Morris <notifications@github.com<mailto:notifications@github.com>> wrote:


alright, got a few more to fix.

I'd be delighted for some help testing this live. I don't know what your process is for using netdev or whatever, last I heard there were some perf tests reaping the whole environment

-

**Comment by tr3buchet** [comment]
yeehaw

**Comment by tr3buchet** [comment]
alright this is fine for now, it needs to be tested once we have a free environment. Manual sg_tests probably before a full regression test on security groups. the `redis_sg_tool.py` especially needs to be tested. may need to wait for dietz on that one

**Comment by asadoughi** [comment]
Have we been able to test this in a free environment? Also, this PR needs to be rebased.

**Comment by tr3buchet** [comment]
alright i've rebased it. if these travis tests pass, it's ready to be tested in netdev.

**Comment by tr3buchet** [comment]
of course they failed

**Comment by tr3buchet** [comment]
anyhow, ready for deploy and fleshing

File: quark/cache/redis_base.py [hunk 1]
```diff
@@ -162,55 +94,66 @@ def vif_key(self, device_id, mac_address):
...
+            values = pipe.execute()
+        return values
 
     @handle_connection_error
     def set_fields(self, keys, field, value):
-        p = self._client.pipeline()
-        for key in keys:
-            p.hset(key, field, value)
-        return p.execute()
+        with self._client.slave.pipeline() as pipe:
```
Comment (asadoughi) [inline quark/cache/redis_base.py]: Should this be master instead of slave?

Comment (tr3buchet) [inline quark/cache/redis_base.py]: yup nice catch!

**Comment by tr3buchet** [comment]
updated for amir's catch

**Comment by asadoughi** [comment]
LGTM

**Pull request merged**